File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int > mp;
        for(int i=0 ; i<nums.size(); i++ ){
            auto it = mp.find(target- nums[i]);
            if(it != mp.end()){
                vector<int> tmp{i, it->second};
                return tmp;
            }
            mp[nums[i]]=i;
        }
        return vector<int>{-1,-1};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        st = {}
        for index in range(0,len(nums)):
            if target - nums[index] in st:
                return [index, st[target-nums[index]]]
            else :
                st[nums[index]]=index
        return [-1,-1]